engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/ejea/PerTestHelper.scala

```diff
@@ -0,0 +1,145 @@
+  }
+
+  def buildEJEA(restarting: Boolean = true,
+                callCachingMode: CallCachingMode = CallCachingOff)
+               (implicit startingState: EngineJobExecutionActorState): TestFSMRef[EngineJobExecutionActorState, EJEAData, MockEjea] = {
+
+    val factory: BackendLifecycleActorFactory = buildFactory()
+    val descriptor = EngineWorkflowDescriptor(backendWorkflowDescriptor, Map.empty, null, null, null, callCachingMode)
+
+    val myBrandNewEjea = new TestFSMRef[EngineJobExecutionActorState, EJEAData, MockEjea](system, Props(new MockEjea(
+      helper = this,
+      jobPreparationProbe = jobPreparationProbe,
+      replyTo = replyToProbe.ref,
+      jobDescriptorKey = jobDescriptorKey,
+      executionData = WorkflowExecutionActorData(descriptor, ExecutionStore(Map.empty), Map.empty, OutputStore(Map.empty)),
+      factory = factory,
+      initializationData = None,
+      restarting = restarting,
+      serviceRegistryActor = serviceRegistryProbe.ref,
+      jobStoreActor = jobStoreProbe.ref,
+      callCacheReadActor = callCacheReadActorProbe.ref,
+      backendName = "NOT USED",
+      callCachingMode = callCachingMode
+    )), parentProbe.ref, s"EngineJobExecutionActorSpec-$workflowId")
+
+    deathwatch watch myBrandNewEjea
+    myBrandNewEjea.setStateInline(state = startingState)
+  }
+}
+
+private[ejea] class MockEjea(helper: PerTestHelper,
+                             jobPreparationProbe: TestProbe,
+                             replyTo: ActorRef,
+                             jobDescriptorKey: BackendJobDescriptorKey,
+                             executionData: WorkflowExecutionActorData,
+                             factory: BackendLifecycleActorFactory,
+                             initializationData: Option[BackendInitializationData],
+                             restarting: Boolean,
+                             serviceRegistryActor: ActorRef,
+                             jobStoreActor: ActorRef,
+                             callCacheReadActor: ActorRef,
+                             backendName: String,
+                             callCachingMode: CallCachingMode) extends EngineJobExecutionActor(replyTo, jobDescriptorKey, executionData, factory, initializationData, restarting, serviceRegistryActor, jobStoreActor, callCacheReadActor, backendName, callCachingMode) {
+
+
+  override def makeFetchCachedResultsActor(cacheHit: CacheHit, taskOutputs: Seq[TaskOutput]) = helper.fetchCachedResultsActorCreations :+= (cacheHit, taskOutputs)
+  override def initializeJobHashing(jobDescriptor: BackendJobDescriptor, activity: CallCachingActivity) = helper.jobHashingInitializations :+= (jobDescriptor, activity)
+  override def createSaveCacheResultsActor(hashes: CallCacheHashes, success: SucceededResponse) = helper.callCacheWriteActorCreations :+= (hashes, success)
```

Comment:
Hmm, probably. That'd make it easier to check we didn't accidentally go x2 on actor creation too. I'll change it